docs/KERNEL-DUMPS.md

```diff
@@ -0,0 +1,48 @@
+# Kernel dump collection in EVE-OS
+
+### Intruduction to kexec and kdump
+
+`kdump` is a standard Linux mechanism to dump machine memory content on kernel crash. `kdump` is based on `kexec`. `kdump` utilizes two kernels: system kernel and dump-capture kernel. System kernel is a normal kernel that is booted with special kdump-specific flags. We need to tell the system kernel to reserve some amount of physical memory where dump-capture kernel will be loaded. We need to load the dump-capture kernel in advance because at the moment crash happens there is no way to read any data from disk because kernel is broken.
+
+Once kernel crash happens the kernel crash handler uses `kexec` mechanism to boot dump-capture kernel. Please note that memory with system kernel is untouched and accessible from dump-capture kernel as seen at the moment of crash. Once dump-capture kernel is booted, the user can use the file `/proc/vmcore` to get access to memory of crashed system kernel. The dump can be saved to disk or copied over network to some other machine for further investigation.
+
+In the EVE environment, the system kernel and the dump-capture kernel are the same. Since the EVE kernel is quite minimal and contains all the options needed for a dump-capture kernel, a single compiled kernel binary simplifies the entire build process and does not require additional disk space to store the capture kernel.
+
+### Setup dump-capture kernel in EVE-OS
+
+First, memory must be reserved for dump capture kernel. The EVE-OS grub configuration has a kernel parameter `crashkernel=128M` which is only specified for the x86_64 architecture. After the system kernel is booted, the `/etc/init.d/000-kexec` script loads the same kernel binary into the reserved area by calling the kexec system tool.
+
+`kexec` loads the dump-capture kernel reusing the same command line with a couple of important differences: `irqpoll nr_cpus=1 reset_devices`, where:
```

Comment:
Nomodule (specified below) is missing in this line

Reply:
Yep.